openedx/core/djangoapps/waffle_utils/tests/test_views.py

```diff
@@ -33,19 +36,89 @@ def test_failure_for_non_staff(self):
     def test_response_with_waffle_switch(self):
-        response = self._get_toggle_state_response(is_staff=True)
+        response = self._get_toggle_state_response()
         self.assertIn('waffle_switches', response.data)
         self.assertTrue(response.data['waffle_switches'])
-        # This is no longer the first switch
-        #self.assertEqual(response.data['waffle_switches'][0]['name'], 'test.switch')
+        waffle_names = [waffle["name"] for waffle in response.data['waffle_switches']]
+        self.assertIn('test.switch', waffle_names)
+
+    def test_response_with_setting_toggle(self):
+        _toggle = SettingToggle("MYSETTING", default=False, module_name="module1")
+        with override_settings(MYSETTING=True):
+            response = self._get_toggle_state_response()
+
+        self.assertIn(
+            {
+                "name": "MYSETTING",
+                "is_active": True,
+                "module": "module1",
+                "class": "SettingToggle",
+            },
+            response.data["django_settings"],
+        )
+
+    def test_response_with_existing_setting_dict_toggle(self):
+        response = self._get_toggle_state_response()
+        self.assertIn(
+            {
+                "name": "FEATURES['MILESTONES_APP']",
+                "is_active": True,
+                "module": "util.milestones_helpers",
+                "class": "SettingDictToggle",
+            },
+            response.data["django_settings"],
+        )
+
+    def test_response_with_new_setting_dict_toggle(self):
+        _toggle = SettingDictToggle(
+            "CUSTOM_FEATURES", "MYSETTING", default=False, module_name="module1"
+        )
+        with override_settings(CUSTOM_FEATURES={"MYSETTING": True}):
+            response = self._get_toggle_state_response()
+
+        setting_dict = {toggle["name"]: toggle for toggle in response.data["django_settings"]}
+
+        self.assertEqual(
+            {
+                "name": "CUSTOM_FEATURES['MYSETTING']",
+                "is_active": True,
+                "module": "module1",
+                "class": "SettingDictToggle",
+            },
+            setting_dict["CUSTOM_FEATURES['MYSETTING']"],
+        )
+
+    def test_setting_overridden_by_setting_toggle(self):
+        _toggle2 = SettingToggle(
+            "MYSETTING2", module_name="module1"
+        )
+        _toggle3 = SettingDictToggle(
+            "MYDICT", "MYSETTING3", module_name="module1"
+        )
+        with override_settings(MYSETTING1=True, MYSETTING2=False, MYDICT={"MYSETTING3": False}):
+            # Need to pre-load settings, otherwise they are not picked up by the view
+            self.assertTrue(settings.MYSETTING1)
+            # self.assertTrue(settings.MYSETTING2)
+            # self.assertEqual({"MYSETTING3": True}, settings.MYDICT)
+            response = self._get_toggle_state_response()
+
+        setting_dict = {toggle["name"]: toggle for toggle in response.data["django_settings"]}
```

Comment:
Can you add an assertion that the size of `response.data["django_settings"]` is 3?

Reply:
Well no, not really, because the response includes all the settings, SettingToggle and SettingDictToggle found in the codebase.

Reply:
I see.  Since the response is a list (I think), I was looking for a way to ensure it didn't include two results with the same name, one for the setting, and one for the SettingToggle, where only the second might be picked up in this test.

Reply:
Well that's simple enough. I'll add something in the test.